src/openfermion/measurements/_bayesian_estimators.py

```diff
@@ -0,0 +1,774 @@
+            if (vectors[2, j]**2 + vectors[1, j]**2) == 0:
+
+                variances.append(maxvar)
+
+            else:
+                variances.append(min(
+                    maxvar,
+                    4 / (vectors[2, j]**2+vectors[1, j]**2) - 1))
+        return variances
+
+    def _make_matrices(self):
+        ''' Function to make the matrices that define multiplication
+        by sin(theta) and cos(theta). We store these as sparse matrices
+        in CSR format, and initialize them in (row,col,data) format.
+        We assume here that self.max_n is less than self.num_freqs.
+        '''
+
+        # self._matrices[n][a] contains matrix M^a(n+1)
+        # following our paper
+        self._matrices = []
+
+        for n in range(1, self._max_n+1):
+
+            # We generate the matrices ourselves in
+            # COO format, and then create a
+            # CSR matrix directly
+
+            cos_matrix, sin_matrix = self._make_matrix(n)
+
+            # Append to list
+            self._matrices.append([cos_matrix, sin_matrix])
+
+    def _make_matrix(self, n):
+
+        # Initialize with terms for when j = 0
+        data_cos = [2]
+        row_ind_cos = [2*n]
+        col_ind_cos = [0]
+
+        data_sin = [-2]
+        row_ind_sin = [2*n-1]
+        col_ind_sin = [0]
+
+        # First do terms when j < n
+        for j in range(1, n):
+
+            data_cos += [1, 1, -1, 1]
+            row_ind_cos += [2*j+2*n-1, 2*j+2*n, 2*n-2*j-1, 2*n-2*j]
+            col_ind_cos += [2*j-1, 2*j, 2*j-1, 2*j]
+
+            data_sin += [1, -1, -1, -1]
+            row_ind_sin += [2*j+2*n, 2*j+2*n-1, 2*n-2*j, 2*n-2*j-1]
+            col_ind_sin += [2*j-1, 2*j, 2*j-1, 2*j]
+
+        # Next do terms when j = n
+        data_cos += [1, 1, 1]
+        row_ind_cos += [4*n-1, 0, 4*n]
+        col_ind_cos += [2*n-1, 2*n, 2*n]
+
+        data_sin += [-1, 1, -1]
+        row_ind_sin += [0, 4*n, 4*n-1]
+        col_ind_sin += [2*n-1, 2*n-1, 2*n]
+
+        # Finish with terms when j > n
+        for j in range(n+1, self._num_freqs+1):
+
+            data_cos += [1, 1]
+            row_ind_cos += [2*j-2*n-1, 2*j-2*n]
+            col_ind_cos += [2*j-1, 2*j]
+
+            data_sin += [-1, 1]
+            row_ind_sin += [2*j-2*n, 2*j-2*n-1]
+            col_ind_sin += [2*j-1, 2*j]
+
+            if j + n <= self._num_freqs:
+
+                data_cos += [1, 1]
+                row_ind_cos += [2*j+2*n-1, 2*j+2*n]
+                col_ind_cos += [2*j-1, 2*j]
+
+                data_sin += [1, -1]
+                row_ind_sin += [2*j+2*n, 2*j+2*n-1]
+                col_ind_sin += [2*j-1, 2*j]
+
+        # Make sparse matrices
+        cos_matrix = sparse.csr_matrix(
+            (data_cos, (row_ind_cos, col_ind_cos)))
+        sin_matrix = sparse.csr_matrix(
+            (data_sin, (row_ind_sin, col_ind_sin)))
+
+        return cos_matrix, sin_matrix
+
+    def _vector_product(self, vectors, round_data):
+
```

Comment:
extra space here. Please run pylint.

Reply:
Pylint didn't catch these spaces suprisingly. I think i've found all of them myself though.